src/Integration/LocalServices/SolutionRuleSetsInformationProvider.cs

```diff
@@ -46,22 +46,38 @@ public SolutionRuleSetsInformationProvider(IServiceProvider serviceProvider)
 
         public IEnumerable<RuleSetDeclaration> GetProjectRuleSetsDeclarations(Project project)
         {
+            /* This method walks through all of the available configurations (e.g. Debug, Release, Foo) and
+             * attempts to fetch the values of a couple of properties from the project (CodeAnalysisRuleSet
+             * and CodeAnalysisRuleSetDirectories). The collected data is put into a data object
+             * and returned to the caller. The collected data includes the DTE Property object itself, which
+             * is used later to update the ruleset value.
+             * 
+             * TODO: consider refactoring. The code seems over-complicated: it finds the "ruleset"
+             * property for all configurations, then backtracks to find the configuration, then looks
+             * for the corresponding "ruleset directories" property.
+             * Note: we are now fetching the "ruleset directories" property from the MSBuild project,
+             * rather than through the DTE (the previous version of this code that used the DTE fails
+             * for C# and VB projects that use the new project system).
+             */
```

Comment:
I've made the smallest fix possible to make the code work for .Net core projects because at this stage a bigger refactoring would be quite risky; I'm not certain exactly how the existing code takes account of the same ruleset being used in multiple configurations, and I don't know how extensive the existing coverage is.
Once we're happy with this change we could refactor with more confidence later.